server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorBalancer.java

```diff
@@ -120,6 +120,11 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
         continue;
       }
 
+      if (strategy.steady(serverHolderList)) {
```

Comment:
for very large clusters this additional computation will be very expensive and cause CPU usage to grind the coordinator to a halt. This is the reason why `emitBalancingStats` is turned off by default, because it becomes computationally intractable for large clusters. Given that this computation is roughly as expensive as `emitBalancingStats`, we should also have a flag to turn this calculation off completely.

If you assume a cluster with identical nodes and roughly the same number of segments per node, then your complexity is O(n^2/k) where n in the number of segments and k the number of nodes. Unfortunately, typically n >> k.